### ISSUE TITLE
trt conversion

### ISSUE THREAD

**Issue by eyebies** [pythonlessons/TensorFlow-2.x-YOLOv3#56]
Thanks for sharing the code! The calibration_input function for int8 passes noise currently. how do we pass the appropriate calibration data?
thanks!

**Comment by pythonlessons** [comment]
Yes, right now it passes random image data. While passing real data it won't change your detection results, so it doesn't matter. Anyway, if you want to pass test or train dataset, you can do that. If I will have time, will do this

**Comment by eyebies** [comment]
Thanks for your response. Here is my calibration function. I see that you do rgb conversion twice (both in the video and image detection functions). Can I skip the conversion?

def calibration_input():
	batch_size = 16
	batched_input = np.zeros((batch_size, YOLO_INPUT_SIZE, YOLO_INPUT_SIZE, 3), dtype=np.float32)

        img_path = 'calib_folder/'
        fnames = glob.glob(img_path + '/*jpg')
        shuffle(fnames)
	for j in fnames[:batch_size]:
		original_image      = cv2.imread(image_path)
		original_image      = cv2.cvtColor(original_image, cv2.COLOR_BGR2RGB)
		image_data = image_preprocess(np.copy(original_image), [YOLO_INPUT_SIZE, YOLO_INPUT_SIZE])
		image_data = image_data[np.newaxis, ...].astype(np.float32)

		batched_input[i, :] = image_data
	batched_input = tf.constant(batched_input)
	yield (batched_input,)

**Comment by pythonlessons** [comment]
Yes, I think you can. 
I think you wrote the code correctly

**Comment by eyebies** [comment]
thanks a lot for your response!

**Comment by Myron1996** [comment]
Hi . I am trying to use TRT when I use 
python3 Convert_to_pb.py It runs successfully 

But when I use python3 Convert_to_TRT.py it gives me following error 

`2020-09-30 11:31:17.138351: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
2020-09-30 11:31:18.131906: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcuda.so.1
2020-09-30 11:31:18.162381: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:982] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2020-09-30 11:31:18.162699: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1716] Found device 0 with properties: 
pciBusID: 0000:01:00.0 name: GeForce GTX 1050 Ti computeCapability: 6.1
coreClock: 1.62GHz coreCount: 6 deviceMemorySize: 3.95GiB deviceMemoryBandwidth: 104.43GiB/s
2020-09-30 11:31:18.162737: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
2020-09-30 11:31:18.162867: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'libcublas.so.10'; dlerror: libcublas.so.10: cannot open shared object file: No such file or directory
2020-09-30 11:31:18.164187: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcufft.so.10
2020-09-30 11:31:18.164444: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcurand.so.10
2020-09-30 11:31:18.165799: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusolver.so.10
2020-09-30 11:31:18.166710: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcusparse.so.10
2020-09-30 11:31:18.169386: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudnn.so.7
2020-09-30 11:31:18.169405: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1753] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
2020-09-30 11:31:18.200406: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'libnvinfer.so.6'; dlerror: libcublas.so.10: cannot open shared object file: No such file or directory
2020-09-30 11:31:18.200428: F tensorflow/compiler/tf2tensorrt/stub/nvinfer_stub.cc:49] getInferLibVersion symbol not found.
Aborted
`
Can you please help me with this error

**Comment by pythonlessons** [comment]
First, next time put your code into ``` at the beginning and end of code, error and etc.
Second, you have an error `Could not load dynamic library 'libnvinfer.so.6';`
This mean TensorRT is not working, check if you installed it. I didn't have TensorRT installation tutorial, because it's quite tricky to install it.
What Cuda version you use? if you use cuda 10.1 and tensorrt 6 try to run following commands:
```
sudo apt-get install libnvinfer6=6.0.1-1+cuda10.1
sudo apt-get install libnvinfer-dev=6.0.1-1+cuda10.1 
sudo apt-get install libnvinfer-plugin6=6.0.1-1+cuda10.1
```
